content/docs/command-reference/exp/apply.md

```diff
@@ -14,20 +14,21 @@ positional arguments:
 
 ## Description
 
-Restores the results of a given `experiment` (name or hash are accepted, see
-`dvc exp run` for details) into the workspace. This means changing the
-appropriate [metafiles](/doc/user-guide/project-structure),
-<abbr>parameter</abbr> files, <abbr>metrics</abbr>, <abbr>plots</abbr>, and
-corresponding DVC-tracked data.
+Restores a given `experiment` (name or hash are accepted, see `dvc exp run` for
+details) into the workspace. This means changing all files (code, data,
+<abbr>parameters</abbr>, <abbr>metrics</abbr>, etc.) needed to reflect the
+experiment conditions and results in the workspace.
 
-> This is similar to `dvc checkout`, but for `dvc experiments`.
+This command is typically used after finding a target `experiment` with
+`dvc exp show` or `dvc exp diff`, and before committing it to Git (making it
+[persistent](/doc/user-guide/experiment-management#persistent-experiments)).
 
-This is typically used after using `dvc exp show` or `dvc exp diff` to find the
-best experiment, and before committing to Git in order to make it
-[persistent](/doc/user-guide/experiment-management#persistent-experiments).
+Note that any [checkpoints](/doc/command-reference/exp/run#checkpoints) found in
```

Comment:
```suggestion
Note that any intermediate [checkpoints](/doc/command-reference/exp/run#checkpoints) found in
```
Wondering if `intermediate` better conveys the message that the whole history is "squashed" into a single merge rather than disregarded. `apply` will still reflect the experiment state at the time of the final checkpoint. Feel free to ignore if you think it's unhelpful.

Reply:
Yeah it's more correct, but not really a concept we have explained much. I don't know if it's intuitive enough. I went for "the history of checkpoints" for now.